[AndroidManifest] Reveal Manifest Changes in PR (`V2`)

ParaskP7 opened the pull request:
<!-- Remember about a good descriptive title. -->

Project Thread: paaHJt-7Tl-p2
Depends On: [a8c-ci-toolkit#164](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/164)
Related: #13582 (`previous workflow`)
Tested By: #13644

<!-- 
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
🛠 Need an AI review?  
Add a comment: `@coderabbitai review`
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
-->

---

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR introduces the `Merged Manifest Diff` CI job that is aiming to help revealing merged manifest changes a PR might introduce.

The main problem this script is trying to address is the fact that:
1. Although explicit `AndroidManifest.xml` changes are visible on a PR review, just by looking at the `Files changed` tab for that specific change, this doesn't necessarily mean that the actual, auto-generated, merged `AndroidManifest.xml` file should have the same amount of changes.
2. Sometimes, implicit `AndroidManifest.xml` changes can occur (think new/updated dependency). Those changes are actually invisible for both, to the author and reviewer of that specific PR.

For more info on the technical/workflow side of things see: p1740739725150069/1740493471.859349-slack-C030U03RC8Y

---

### Review

@malinajirka I am adding you as an additional reviewer to @wzieba just so that at least one product engineer is aware of this new CI workflow and CI check. I hope this will, through you, give a decent heads-up to other product engineers as well.

FYI: At a later point of time a formal announcement will be posted.

---

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

You could check the associated test PR (#13644) and see when and how such comments might appear ([here](https://github.com/woocommerce/woocommerce-android/pull/13644#issuecomment-2690635509) and [here](https://github.com/woocommerce/woocommerce-android/pull/13644#issuecomment-2690636426)):

![image](https://github.com/user-attachments/assets/f5137241-e111-47c4-8d23-45f0d6d888dd)

FYI: The above test comment appeared due to an implicit `AndroidManifest.xml` change, adding the [singular](https://support.singular.net/hc/en-us/articles/360037581952-Android-SDK-Integration-Guide) dependency to the project. It would have been impossible for an engineer to know how the merged `AndroidManifest.xml` file got updated without doing such a check manually.

---

### Merge Instructions

- [x] Wait for [a8c-ci-toolkit#164](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/164) to be reviewed, approved and merged.
- [x] Create a new [a8c-ci-toolkit release](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/releases) ([3.10.1](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/releases/tag/3.10.1) [current] -> [5.1.0](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/releases/tag/5.1.0) [merge manifest diff] -> [5.1.2](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/releases/tag/5.1.2) [latest]).
- [x] Update [CI_TOOLKIT](https://github.com/woocommerce/woocommerce-android/blob/trunk/.buildkite/shared-pipeline-vars#L6) to point to this newly created [a8c-ci-toolkit release](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/releases) (5cd93fe9d7a8f655239896c322d2eee6d02b6437).
- [x] Remove `status: do not merge` label.
- [x] Merge this PR.
- [x] Close #13644 and #13662 test-only PRs.

---

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

ParaskP7 committed:
CI: Update a8c-ci-toolkit to pr-sha commit (merged manifest diff)

FYI: This change is done for testing purposes and until the below
'A8C CI Toolkit' #164 PR gets merged to 'trunk'. When that's
done, the 'a8c-ci-toolkit' will be updated to '4.1.0' instead.

A8C CI Toolkit PR: [AndroidManifest] Merged Manifest Diff #164
- https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/164

ParaskP7 committed:
CI: Add merged manifest diff

This new CI job will be responsible to detect and report changes in
merged 'AndroidManifest.xml' files during PR reviews.

FYI: Since WCAndroid is having two 'app' related modules, 'WooCommerce'
and 'WooCommerce-Wear', the merged manifest diff will be triggered
separately and for both of them.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ParaskP7 committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-android into manifest/reveal-changes-in-pr-v2

ParaskP7 commented:
@coderabbitai review

ParaskP7 committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-android into manifest/reveal-changes-in-pr-v2

ParaskP7 committed:
CI: Update a8c-ci-toolkit to 5.1.1

Release Notes: https://github.com/Automattic/
a8c-ci-toolkit-buildkite-plugin/releases/tag/5.1.1

Related: 396dc7e42f875e43c6be3d8c9d5519261be82969

ParaskP7 committed:
CI: Update a8c-ci-toolkit to 5.1.2

Release Notes: https://github.com/Automattic/
a8c-ci-toolkit-buildkite-plugin/releases/tag/5.1.2

FYI: This is done to include the newly added TL;DR change as part of
this PR, just to avoid creating another PR just for that matter.
